### PR TITLE
add phalcon psr supported

### PR DESCRIPTION
**Pull request by kenxx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | none
| Documentation   | none
| License         | MIT

#### What's in this PR?

Add `Phalcon\Http\Message\*Factory` to `src/Strategy/CommonPsr17ClassesStrategy.php`

#### Why?

Cause when I use sentry in phalcon 4.0, they said some factory not found, but phalcon has those factories.

#### Example Usage

nothing changes

#### Checklist

- [ ] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

#### To Do

- [ ] If the PR is not complete but you want to discuss the approach, list what remains to be done here

### CODE REVIEW

**Comment by dbu** [comment]
cool that phalcon supports PSR-17!

how difficult is it to install phalcon in the CI? any chance you could add it in https://github.com/php-http/discovery/blob/4739e2675c150f0aa94421c76c7c403703dcc1d5/.travis.yml#L63 so that we have a CI test for it?

**Comment by dbu** [comment]
i fixed the styleci configuration issue in master, please rebase this branch on master to get that build fixed.

File: tests/install.sh [hunk 1]
```diff
@@ -21,7 +21,14 @@ composer init --working-dir $BUILD_DIR --no-interaction
 composer req --working-dir $BUILD_DIR php-http/discovery --no-update
 
 # Define packages from arguments
-composer req --working-dir $BUILD_DIR $3
+if ! [ -z "$3" ]; then
+    composer req --working-dir $BUILD_DIR $3
+fi
+
+# Arg 4 means some pecl things will be install
```
Comment (dbu) [inline tests/install.sh:28]:
```suggestion
# Argument 4 allows to install pecl dependencies
```

File: tests/install.sh [hunk 2]
```diff
@@ -21,7 +21,14 @@ composer init --working-dir $BUILD_DIR --no-interaction
 composer req --working-dir $BUILD_DIR php-http/discovery --no-update
 
 # Define packages from arguments
```
Comment (dbu) [inline tests/install.sh:23]:
```suggestion
# Argument 3 installs additional composer packages
```

**Review by dbu — APPROVED** [review]
this looks good, thanks!

i have some nitpicks on the install script doc comments if you don't mind. otherwise unless the build fails, looks good to merge

**Comment by kenxx** [comment]
i not very familiar with travis ci, sorry about that.

i went to see their docs, it looks kind of easy, so i tried to write some.

feel free to correct my codes, thanks for help.

**Comment by kenxx** [comment]
oh i forget to echo "extension=*.so" to the script, but now it looks good.

there is one more small things is that any way to set the *.so to cache files?

**Comment by kenxx** [comment]
@dbu turns out there is no need to add .so to php.ini using pecl in TravisCI, that's really good.

**Comment by dbu** [comment]
something seems to fail on travis-ci with the vendors not being found: https://travis-ci.org/github/php-http/discovery/jobs/723715146

**Comment by kenxx** [comment]
@dbu finally!

File: tests/install.sh [hunk 3]
```diff
@@ -18,15 +18,26 @@ mkdir -p $BUILD_DIR
...
 composer req --working-dir $BUILD_DIR php-http/discovery --no-update
 
-# Define packages from arguments
-composer req --working-dir $BUILD_DIR $3
+# Argument 3 installs additional composer packages
+if ! [ -z "$3" ]; then
+    composer req --working-dir $BUILD_DIR $3
+else
+    # composer req --no-update impiles --no install
+    composer req --working-dir $BUILD_DIR
```
Comment (dbu) [inline tests/install.sh:29]: isn't this the same as above when `$3` is `""`? can we simply drop the if/else and revert to the previous code?

Comment (kenxx) [reply tests/install.sh:29]: haha, that's good, i just focus on how to fix that error.

Comment (dbu) [reply tests/install.sh:29]: that is why we do reviews ;-) 
getting it to run was the hard work!

Comment (kenxx) [reply tests/install.sh]: looks good

**Review by dbu — APPROVED** [review]
awesome!

i will merge on the cli to squash away the testing commits and tag a new version

**Comment by dbu** [comment]
merged and tagged as 1.10.0. thanks a lot!

**Pull request closed without merging**